addons/stock/models/stock_move_line.py

```diff
@@ -142,12 +142,11 @@ def _compute_quantity(self):
         for record in self:
             if not record.quant_id or record.quantity:
                 continue
-            if float_compare(record.move_id.product_qty, record.quantity, precision_rounding=record.move_id.product_uom.rounding) > 0:
+            if record.move_id and float_compare(record.move_id.product_qty, record.move_id.quantity, precision_rounding=record.move_id.product_uom.rounding) > 0:
                 qty = max(0, min(record.quant_id.available_quantity, record.move_id.product_qty - record.move_id.quantity))
```

Comment:
```suggestion
                quantity_in_product_uom = record.move_id.product_uom._compute_quantity(move.quantity, move.product_id.uom_id)
                qty = max(0, min(record.quant_id.available_quantity, record.move_id.product_qty - quantity_in_product_uom))
```
This should probably be something like this, so all quantities are compared in the ***product***'s uom. This needs to be tested though.